### PR TITLE
Fix incorrect return check of BN_bn2binpad

### PR DESCRIPTION
**Pull request by PeiweiHu**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/ec/ec_deprecated.c [hunk 1]
```diff
@@ -52,7 +52,7 @@ EC_POINT *EC_POINT_bn2point(const EC_GROUP *group,
         return NULL;
     }
 
-    if (!BN_bn2binpad(bn, buf, buf_len)) {
+    if (BN_bn2binpad(bn, buf, buf_len) <= 0) {
```
Comment (kroeckx) [inline crypto/ec/ec_deprecated.c:55]: I think 0 is actually valid, please use < 0

Comment (PeiweiHu) [reply crypto/ec/ec_deprecated.c]: thanks, already changed.

File: test/acvp_test.c [hunk 2]
```diff
@@ -71,7 +71,7 @@ static int pkey_get_bn_bytes(EVP_PKEY *pkey, const char *name,
     buf = OPENSSL_zalloc(sz);
     if (buf == NULL)
         goto err;
-    if (!BN_bn2binpad(bn, buf, sz))
+    if (BN_bn2binpad(bn, buf, sz) <= 0)
```
Comment (kroeckx) [inline test/acvp_test.c:74]: Same problem here

File: test/ecdsatest.c [hunk 3]
```diff
@@ -46,7 +46,7 @@ static int fbytes(unsigned char *buf, size_t num, ossl_unused const char *name,
         || !TEST_true(BN_hex2bn(&tmp, numbers[fbytes_counter]))
         /* tmp might need leading zeros so pad it out */
         || !TEST_int_le(BN_num_bytes(tmp), num)
-        || !TEST_true(BN_bn2binpad(tmp, buf, num)))
+        || !TEST_int_gt(BN_bn2binpad(tmp, buf, num), 0))
```
Comment (kroeckx) [inline test/ecdsatest.c:49]: Same problem here

File: test/acvp_test.c (hunk 2 above)
Comment (PeiweiHu) [reply test/acvp_test.c:74]: I changed to <= according to reutrn values check in /test/acvp_test.c:229, 231, 570, 572. There are a lot. Do we need to change them all?

Comment (PeiweiHu) [reply test/acvp_test.c:74]: And /crypto/ec/ec_curve.c: 3404

Comment (t8m) [reply test/acvp_test.c:74]: > Same problem here

Hmm, @kroeckx  but what the returned 0 indicates? A 0 byte sized number does not look like a number. Perhaps 0 cannot be returned at all? A zero BIGNUM should be encoded as a single zero byte, shouldn't it? So the return value should be 1.

**Comment by kroeckx** [comment]
> Hmm, @kroeckx  but what the returned 0 indicates? A 0 byte sized number does not look like a number. Perhaps 0 cannot be returned at all? A zero BIGNUM should be encoded as a single zero byte, shouldn't it? So the return value should be 1.

I guess you only get 0 back when the number is 0 and the size that is passed is 0, which seems unlikely. So <= 0 is fine.

File: test/acvp_test.c (hunk 2 above)
Comment (paulidale) [reply test/acvp_test.c:74]: It's hard to see how a zero could be returned.

We should nonetheless try to be consistent in the return code handling.

**Review by paulidale — APPROVED** [review]

File: test/acvp_test.c (hunk 2 above)
Comment (PeiweiHu) [reply test/acvp_test.c:74]: In https://github.com/openssl/openssl/pull/16943, the code invoking BN_bn2nativepad has a consistent handling way. As for  BN_bn2binpad, == -1, <0, and <=0 can be found in codebase.

Comment (paulidale) [reply test/acvp_test.c:74]: Agreed, they should be unified.  Not necessary for this PR IMO.

**Review by kroeckx — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master and 3.0.  Thanks for the fix.

**Pull request closed without merging**